crates/extension_api/src/settings.rs

```diff
@@ -21,6 +45,106 @@ impl LspSettings {
     }
 }
 
+/// Information about an LSP server binary.
+#[derive(Debug, Default)]
+pub struct LspServerInfo {
+    /// The name of the binary to search for in the PATH.
+    pub binary_name: Option<String>,
+    /// The path to the cached binary from a previous automatic download.
+    pub cached_binary_path: Option<String>,
+    /// The default arguments to pass to the binary if both the binary and arguments are not set.
+    pub default_arguments: Vec<String>,
+}
+
+impl Worktree {
+    /// Creates a [`wit::Command`] from the given [`CommandSettings`] and path. The environment variables are merged
+    /// with the shell environment.
+    pub fn create_command(&self, command: &CommandSettings, path: String) -> wit::Command {
+        let (platform, _) = crate::current_platform();
+        let env = match platform {
+            Os::Mac | Os::Linux => self.shell_env(),
+            Os::Windows => Default::default(),
+        };
+
+        let env = env
+            .into_iter()
+            .chain(
+                command
+                    .env
+                    .iter()
+                    .flatten()
+                    .map(|(k, v)| (k.clone(), v.clone())),
+            )
+            .collect();
+
+        wit::Command {
+            command: path,
+            args: command.arguments.clone().unwrap_or_default(),
+            env,
+        }
+    }
+
+    /// Searches for a language server binary.
+    ///
+    /// This function will search for the binary in the PATH, then check the configured binary path, and finally check
+    /// the cached binary path.
+    pub fn find_language_server(
+        &self,
+        settings: &LspSettings,
+        info: &LspServerInfo,
+    ) -> LanguageServerPath {
+        let mut path = None;
+        if let Some(binary_name) = settings
+            .allow_path_search
+            .unwrap_or(true)
+            .then_some(info.binary_name.as_deref())
+            .flatten()
+        {
+            path = check_exists(self.which(binary_name));
+        }
+
+        path = path.or_else(|| check_exists(settings.binary.as_ref().and_then(|v| v.path.clone())));
+
+        let is_cached = path.is_none();
+        if settings.allow_automatic_download.unwrap_or(true) {
+            path = path.or_else(|| check_exists(info.cached_binary_path.clone()));
+        }
+
+        let default = CommandSettings::default();
+        let mut command = path.map(|command| {
+            self.create_command(settings.binary.as_ref().unwrap_or(&default), command)
+        });
+
+        if settings
```

Comment:
I believe that this behavior would be the most expected. If you specify either `path` or `args` in the settings you will disable the defaults for _both._